collector/benchmarks/README.md

```diff
@@ -143,7 +143,88 @@ Rust code being written today.
 - **regex**: See above. This is an older version of the crate.
 - **piston-image**: See above. This is an older version of the `image` crate.
 - **style-servo**: An old version of Servo's `style` crate. A large crate, and
-  one used by old versions of Firefox.
+  one used by old versions of Firefox. Built with `--features=gecko`.
 - **syn**: See above. This is an older version (0.11.11) of the crate.
 - **tokio-webpush-simple**: A simple web server built with a very old version
   of tokio. Uses futures a lot, but doesn't use `async`/`await`.
+
+# How to update/add/remove benchmarks
+
+## Add a new benchmark
+
+- Decide on which category it belongs to.
+- If it's a third-party crate:
+  - If you are keen: talk with a maintainer of the crate to see if there is
+    anything we should be aware of when using this crate as a compile-time
+    benchmark.
+  - Look at [crates.io](https://crates.io) to find the latest (non-prerelease) version.
+  - Download it with `collector download -c $CATEGORY crate $NAME $VERSION`.
+    The `$CATEGORY` is probably `primary`.
+- It makes it easier for reviewers if you split things into two commits.
+- In the first commit, just add the new code.
+  - Do this by doing `git add` on the new directory.
+- In the second commit, do everything else.
+  - Add `[workspace]` to the very bottom of the benchmark's `Cargo.toml`, if
+    doesn't already have a `[workspace]` section. This means commands like
+    `cargo build` will work within the benchmark directory.
+  - Add any necessary stuff to the `perf-config.json` file.
+    - If the benchmark is a sub-crate within a top-level crate, you'll need a
+      `"cargo_toml"` entry.
+    - If you get a "non-wrapped rustc" error when running it, you'll need a
+      `"touch_file"` entry.
+  - Consider adding one or more `N-*.patch` files.
+    - If it's a primary benchmark, you should definitely do this.
+    - Creating the diff against what you've committed so far might be useful.
+      Use `git diff` from the repository root, or `git diff --relative` within
+      the benchmark directory. Copy the output into the `N-*.patch` file.
+    - Do a test run with an `IncrPatched` scenario to make sure the patch
```

Comment:
Should we say how to do a local run here? This has the potential to get out of sync with other docs, but might be helpful:

```
target/release/collector bench_local +nightly --id Test --profiles=Check --scenarios=IncrPatched --include=$NEW_BENCHMARK
```